packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/ContiguousScopeHandler.ts

```diff
@@ -0,0 +1,191 @@
+
+export class ContiguousScopeHandler extends BaseScopeHandler {
+  protected readonly isHierarchical = false;
+  private readonly scopeHandler: ScopeHandler;
+
+  constructor(
+    private scopeHandlerFactory: ScopeHandlerFactory,
+    public scopeType: ContiguousScopeType,
+    languageId: string,
+  ) {
+    super();
+    const handler = scopeHandlerFactory.create(scopeType.scopeType, languageId);
+    if (handler == null) {
+      throw new Error(
+        `No available scope handler for '${scopeType.scopeType.type}'`,
+      );
+    }
+    this.scopeHandler = handler;
+  }
+
+  get iterationScopeType(): ScopeType | CustomScopeType {
+    return this.scopeHandler.iterationScopeType;
+  }
+
+  generateScopeCandidates(
+    editor: TextEditor,
+    position: Position,
+    direction: Direction,
+  ): Iterable<TargetScope> {
+    return direction === "backward"
+      ? this.generateScopeCandidatesBackward(editor, position)
+      : this.generateScopeCandidatesForward(editor, position);
+  }
+
+  *generateScopeCandidatesBackward(
+    editor: TextEditor,
+    position: Position,
+  ): Iterable<TargetScope> {
+    let targetsForward = next(
+      getTargetsInDirection(this.scopeHandler, editor, position, "forward"),
+    );
+
+    const targetsBackward = getTargetsInDirection(
+      this.scopeHandler,
+      editor,
+      position,
+      "backward",
+    );
+
+    for (const targets of targetsBackward) {
+      if (targetsForward != null && isAdjacent(targets[1], targetsForward[0])) {
+        yield targetsToScope(targets[0], targetsForward[1]);
+        targetsForward = undefined;
+      } else {
+        yield targetsToScope(...targets);
+      }
+    }
+  }
+
+  *generateScopeCandidatesForward(
+    editor: TextEditor,
+    position: Position,
+  ): Iterable<TargetScope> {
+    let targetsBackward = next(
+      getTargetsInDirection(this.scopeHandler, editor, position, "backward"),
+    );
+
+    const targetsForward = getTargetsInDirection(
+      this.scopeHandler,
+      editor,
+      position,
+      "forward",
+    );
+
+    for (const targets of targetsForward) {
+      if (
+        targetsBackward != null &&
+        isAdjacent(targetsBackward[1], targets[0])
+      ) {
+        yield targetsToScope(targetsBackward[0], targets[1]);
+        targetsBackward = undefined;
+      } else {
+        yield targetsToScope(...targets);
+      }
+    }
+  }
+}
+
+function targetsToScope(
+  leadingTarget: Target,
+  trailingTarget: Target,
+): TargetScope {
+  if (leadingTarget.contentRange.isRangeEqual(trailingTarget.contentRange)) {
+    return {
+      editor: leadingTarget.editor,
+      domain: leadingTarget.contentRange,
+      getTargets: () => [leadingTarget],
+    };
+  }
+
+  const range = leadingTarget.contentRange.union(trailingTarget.contentRange);
+  return {
+    editor: leadingTarget.editor,
+    domain: range,
+    getTargets: () => [leadingTarget.withContentRange(range)],
```

Comment:
why not use our new `createContinuousRangeTarget` function?

Reply:
Because I needed the continuous range for the domain before we even create the target

Reply:
Not sure I follow. Why can't you use union for domain and `createContinuousRangeTarget` for the target?

Reply:
I meant in this case where content range and domain is this same I don't want to have that calculation in two separate places. If we do a separate calculation for domain its no problem.

Reply:
I don't mean where content range and domain is the same between two scopes. I mean the resulting content range and domain for the new scope is the same. We're lazily constructing the targets in a callback and I needed the content range to even create the scope in the first place

Reply:
Anyhow I now use `createContinuousRangeTarget `